lightning/src/events/mod.rs

```diff
@@ -1702,7 +1707,9 @@ impl Writeable for Event {
 					Some(PaymentFailureReason::InvoiceRequestRejected) =>
 						&Some(PaymentFailureReason::RecipientRejected),
 					Some(PaymentFailureReason::BlindedPathCreationFailed) =>
-						&Some(PaymentFailureReason::RouteNotFound)
+						&Some(PaymentFailureReason::RouteNotFound),
+					Some(PaymentFailureReason::InvalidAmount) =>
+						&Some(PaymentFailureReason::InvalidAmount)
```

Comment:
Won't this break downgrades? We need to map to a variant available prior to version 0.0.124. I guess it is never serialized in an event, and instead return directly? I wonder if we should make a `Bolt11PaymentError` similar to `Bolt12PaymentError` instead.